internal/cloudapi/v2/imagerequest.go

```diff
@@ -47,10 +47,8 @@ func newAWSTarget(options UploadOptions, imageType distro.ImageType) (*target.Ta
 		return nil, HTTPError(ErrorJSONUnMarshallingError)
 	}
 
-	// For service maintenance, images are discovered by the "Name:composer-api-*"
-	// tag filter. Currently all image names in the service are generated, so they're
-	// guaranteed to be unique as well. If users are ever allowed to name their images,
-	// an extra tag should be added.
+	// For service maintenance, images are discovered by the "Name:composer-api-*" tag filter. The `Name` tag differs
+	// from the AMI Name. The later being user customizable.
```

Comment:
The previous version of this comment said that if we ever allow users to name their images, we should add an extra tag.  The way I read it is that we should tag images with something composer-specific for maintenance and set the name to what the user wanted.  @croissanne?  (I think you wrote the original comment).

I think that would make more sense to set the name to what the user provided, but since you mentioned the tags aren't shared, and the "Name" field (in your screenshot) is controlled by the Name tag (is that right?), then maybe it's fine to just name the AMI and leave the rest the same.

Reply:
Maybe explicitly separating them makes sense for clarity, but there's no functional difference.

 @achilleas-k when I wrote the comment I didn't realise names are inferred from the tags, and thus aren't shared. So we're stuck with setting the ami name (which has to be unique).